File: spray-can-only/src/main/scala/com/example/Book.scala
package com.example

import com.github.nscala_time.time.Imports._
import spray.json._
import DefaultJsonProtocol._

case class Book(title: String, authors: List[String], publisher: String, isbn13: Long, publishedDate: String)

object MyJsonProtocol extends DefaultJsonProtocol {
  implicit val bookFormat = jsonFormat5(Book.apply)
}

object Book {
  private var books = 
    List(
        Book("Animal Farm", List("George Orwell"), "Penguin", 9780141036137L, DateTimeFormat.forPattern("MMMM d, yyyy").print(DateTime.now)),
        Book("1984", List("George Orwell"), "Penguin", 9780141036144L, DateTimeFormat.forPattern("MMMM d, yyyy").print(DateTime.now)),
        Book("The Razor's Edge", List("William Sommerset Maugham"), "Vintage", 9780099284864L, DateTimeFormat.forPattern("MMMM d, yyyy").print(DateTime.now))
    )
    
    def all = books 
}

File: dropwizard/src/main/java/Book.java
public class Book {

    private final String title;
    private final String[] authors;
    private final String publisher;
    private final long isbn13;
    private final String publishedDate;

    public Book(String title, String[] authors, String publisher, long isbn13, String publishedDate) {
        this.title = title;
        this.authors = authors;
        this.publisher = publisher;
        this.isbn13 = isbn13;
        this.publishedDate = publishedDate;
    }

    public String getTitle() {
        return title;
    }

    public String[] getAuthors() {
        return authors;
    }

    public String getPublisher() {
        return publisher;
    }

    public long getIsbn13() {
        return isbn13;
    }

    public String getPublishedDate() {
        return publishedDate;
    }
}